### PR TITLE
Move all internal types from /snap-types to /snap-utils

### LINKED ISSUE #693: Move internal types from /snap-types to /snap-utils
`@metamask/snap-types` should only contain types that are needed during snap development. All other, "internal" types should either live by their logical home / origin / concrete implementation or in `@metamask/snap-utils`.

### PR DESCRIPTION
**Pull request by rekmarks**
Moves all "internal" types from `@metamask/snap-types` to `@metamask/snap-utils`. `/snap-types` now only contains types that are used by snaps themselves. All other types should go into `@metamask/snap-utils` or alongside their concrete implementations sake of convenience.

Fixes #693

### CODE REVIEW

File: packages/controllers/src/snaps/SnapController.ts [hunk 1]
```diff
@@ -1718,7 +1719,7 @@ export class SnapController extends BaseController<
     }
   }
 
-  private async _startSnap(snapData: SnapData) {
+  private async _startSnap(snapData: { snapId: string; sourceCode: string }) {
```
Comment (Mrtenz) [inline packages/controllers/src/snaps/SnapController.ts:1722]: Why did you remove this type?

Comment (rekmarks) [reply packages/controllers/src/snaps/SnapController.ts:1722]: Because its name was vague, collided with completely different `SnapData` type in execution environments, and (most importantly) wasn't used anywhere but here.

**Review by Mrtenz — APPROVED** [review]
